### PR TITLE
Enable publish cmdlets on PWSH and Windows Nano Server

### LINKED ISSUE #25: Enable Publish-Module and Publish-Script cmdlets on PowerShell Core Edition
_From @raghushantha on August 8, 2016 18:50_

These cmdlets in PowerShellGet module take dependency on nuget.exe which is windows full sku specific

We need to rework the logic to use Nuget Client APIs so that the publish-\* cmdlets work on CorePowerShell

_Copied from original issue: PowerShell/PowerShell#1678_

### PR DESCRIPTION
**Pull request by bmanikm**
- Enabled publish cmdlets on PWSH and Windows Nano Server
  - Dotnet command version 2.0.0 or newer should be installed by the user prior to using the publish cmdlets on PWSH and Windows Nano Server.
  - Users can install the dotnet command by following the instructions specified at https://aka.ms/dotnet-install-script .
  - On Windows, users can install the dotnet command by running *Invoke-WebRequest -Uri 'https://dot.net/v1/dotnet-install.ps1' -OutFile '.\dotnet-install.ps1'; & '.\dotnet-install.ps1' -Channel Current -Version '2.0.0'*
  - Added checks for dotnet command availability.
    - Added additional check for dotnet cmd version. 2.0.0 or higher is required.
  - Updated the prompt logic for NuGet.exe to include dotnet command availability
  - Enabled existing publish tests on PWSH
  - Validated the changes on Nano Server
  - Added test case for validating error when dotnet command is not available on Linux, MacOS.
  - Enabled the publish tests for prerelease and accept license features on PWSH.
- PWSH: Changed the install location of AllUsers scope on PWSH to SHARED_MODULES location ($PSHOME\..\Modules and $PSHOME\..\Scripts).
- Updated the logic for detecting Windows Nano Server platform.
- Added Join-PathUtility to get the case-sensitve paths on Linux for License acceptance tests, etc.
- Additional change to support installation of items with 2 part version published using dotnet command.
- Enabled dotnet command usage on Windows PowerShell.
  - Publish cmdlets on Windows PowerShell supports using the dotnet command for publishing operations.
- Updated the change log and release notes.
- Changed the version of PowerShellGet module to 1.6.0.

Resolves #25.

### CODE REVIEW

File: PowerShellGet/PSGet.Resource.psd1 [hunk 1]
```diff
@@ -76,8 +76,9 @@ ConvertFrom-StringData @'
...
+        CouldNotFindDotnetCommand=For publish operations, dotnet command version '{0}' or newer is required to interact with the NuGet-based repositories. Please ensure that dotnet command version '{0}' or newer is installed and available under one of the paths specified in PATH environment variable value. You can also install the dotnet command by following the instructions specified at '{1}'.
+        CouldNotInstallNuGetBinaries2=PowerShellGet requires NuGet.exe (or dotnet command) and NuGet provider version '{0}' or newer to interact with the NuGet-based repositories. Please ensure that '{0}' or newer version of NuGet provider is installed and NuGet.exe (or dotnet command) is available under one of the paths specified in PATH environment variable value.        
         InstallNuGetProviderShouldContinueQuery=PowerShellGet requires NuGet provider version '{0}' or newer to interact with NuGet-based repositories. The NuGet provider must be available in '{1}' or '{2}'. You can also install the NuGet provider by running 'Install-PackageProvider -Name NuGet -MinimumVersion {0} -Force'. Do you want PowerShellGet to install and import the NuGet provider now?
```
Comment (farehar) [inline PowerShellGet/PSGet.Resource.psd1:82]: Please have these messages reviewed by Keith. #Closed

Comment (bmanikm) [reply PowerShellGet/PSGet.Resource.psd1:82]: @JKeithB Please review these messages. Thanks. #Closed

File: PowerShellGet/PSModule.psm1 [hunk 2]
```diff
@@ -135,6 +160,10 @@ $script:NuGetClientSourceURL = 'https://go.microsoft.com/fwlink/?LinkID=690216&c
 $script:NuGetExeName = 'NuGet.exe'
 $script:NuGetExePath = $null
 $script:NuGetProvider = $null
+$script:DotnetCommandName = 'dotnet'
+$script:MinimumDotnetCommandVersion = [Version]'2.0.0'
+$script:DotnetInstallUrl = 'https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script'
```
Comment (farehar) [inline PowerShellGet/PSModule.psm1]: >'https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script' [](start = 27, length = 74)

should this be a fwdlink? #Closed

File: Tests/PSGetPublishModule.Tests.ps1 [hunk 3]
```diff
@@ -179,12 +179,13 @@ Describe PowerShell.PSGet.PublishModuleTests -Tags 'BVT','InnerLoop' {
...
     It "PublishModuleWithPath" {
         $version = "1.0"
-        New-ModuleManifest -Path "$script:PublishModuleBase\$script:PublishModuleName.psd1" -ModuleVersion $version -Description "$script:PublishModuleName module"  -NestedModules "$script:PublishModuleName.psm1"
+        New-ModuleManifest -Path (Join-Path -Path $script:PublishModuleBase -ChildPath "$script:PublishModuleName.psd1") -ModuleVersion $version -Description "$script:PublishModuleName module"  -NestedModules "$script:PublishModuleName.psm1"
         Publish-Module -Path $script:PublishModuleBase -NuGetApiKey $script:ApiKey -ReleaseNotes "$script:PublishModuleName release notes" -Tags PSGet -LicenseUri "https://$script:PublishModuleName.com/license" -ProjectUri "https://$script:PublishModuleName.com" -WarningAction SilentlyContinue
         $psgetItemInfo = Find-Module $script:PublishModuleName -RequiredVersion $version
         Assert ($psgetItemInfo.Name -eq $script:PublishModuleName) "Publish-Module should publish a module with valid module path, $($psgetItemInfo.Name)"
     }
 
+    # Purpose: Publish a module with -Path
```
Comment (farehar) [inline Tests/PSGetPublishModule.Tests.ps1]: remove this. #Closed

**Review by farehar — APPROVED** [review]
:shipit:

**Pull request merged**